tools/Azure.Mcp.Tools.EventHubs/src/Services/EventHubsService.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Mcp.Core.Options;
+using Azure.Mcp.Core.Services.Azure;
+using Azure.Mcp.Core.Services.Azure.Subscription;
+using Azure.Mcp.Core.Services.Azure.Tenant;
+using Azure.Mcp.Tools.EventHubs.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.EventHubs.Services;
+
+public class EventHubsService(ISubscriptionService subscriptionService, ITenantService tenantService, ILogger<EventHubsService> logger)
+    : BaseAzureResourceService(subscriptionService, tenantService), IEventHubsService
+{
+    private readonly ISubscriptionService _subscriptionService = subscriptionService;
+    private readonly ILogger<EventHubsService> _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+    public async Task<List<EventHubsNamespaceInfo>> GetNamespacesAsync(
+        string resourceGroup,
+        string subscription,
+        string? tenant = null,
+        RetryPolicyOptions? retryPolicy = null)
+    {
+        return await ExecuteResourceQueryAsync(
+            "Microsoft.EventHub/namespaces",
+            resourceGroup,
+            subscription,
+            retryPolicy,
+            ConvertToEventHubsNamespaceInfo);
+    }
+
+    private static EventHubsNamespaceInfo ConvertToEventHubsNamespaceInfo(JsonElement item)
+    {
+        Models.EventHubsNamespaceData? eventHubsNamespace = Models.EventHubsNamespaceData.FromJson(item);
+        if (eventHubsNamespace == null)
+        {
+            throw new InvalidOperationException("Failed to parse EventHubs namespace data");
+        }
+
+        if (string.IsNullOrEmpty(eventHubsNamespace.ResourceId))
+            throw new InvalidOperationException("Resource ID is missing");
+        var id = new ResourceIdentifier(eventHubsNamespace.ResourceId);
+
+        return new EventHubsNamespaceInfo(
+            Name: eventHubsNamespace.ResourceName ?? "Unknown",
```

Comment:
I don't think you need to perform a ?? if you've already done that null check above.  None of these fields should be optional.